### PR TITLE
Fix for possible OOB writes when running the specificquery client

### PR DESCRIPTION
**Pull request by kjsanger**
Add a bounds check for the maximum permitted number of query parameters.

Add a test for exceeding the maximum.

### CODE REVIEW

File: src/query.c [hunk 1]
```diff
@@ -751,17 +802,23 @@ query_format_in_t* make_query_format_from_sql(const char *sql) {
 error_recoverable:
```
Comment (mksanger) [inline src/query.c:802]: This isn't recoverable if it just goes to the same place as other errors

Comment (kjsanger) [reply src/query.c:802]: Yes, all it does is an additional log statement. I left that alone intentionally to be tackled in its own PR because it's not related to this bug.
